Remove hardcoded InputTag in PATPhotonProducer

matteosan1 opened the pull request:
The hardcoded inputtag were preventing the electronVeto in miniAOD to work properly. Different superClusterRefs for (slimmedPhotons and gedGsfElectrons) were indeed compared returning identically false.
Changed accordingly the configuration to produce miniAOD.

arizzi committed:
store rapidity instead of pseudorapidity for PackedGenCandidates

arizzi committed:
add class version and adapt producer, adding ioread to keep compatibility

Resolved Conflicts:
	DataFormats/PatCandidates/src/classes_def_objects.xml

bendavid committed:
first implementation of MCTruthHelper and storage of GenStatusFlags in GenParticles

bendavid committed:
make helper functions work with both HepMC::GenPartices and reco::GenParticles, add accessors and additional flags for hard process taus

bendavid committed:
fix double counting of hard process particles

bendavid committed:
add dictionary for std::bitset<13> (used in GenStatusFlags)

bendavid committed:
add some additional flags and fix handling of radiated photons in pre-HepMC standard generators

bendavid committed:
revise selection of flags

bendavid committed:
add GenStatusFlags also to PackedGenParticle

bendavid committed:
add classversion and checksum for PackedGenParticle

bendavid committed:
actually fill statusFlags in GenParticlePruner

bendavid committed:
Merged genStatusFlags from repository bendavid

wmtan committed:
File information in playback info

matteosan1 committed:
remove hardcoded inputtag

cmsbuild commented:
A new Pull Request was created by @matteosan1 (Matteo Sani) for CMSSW_7_4_X.

Remove hardcoded InputTag in PATPhotonProducer

It involves the following packages:

PhysicsTools/PatAlgos


@cmsbuild, @vadler, @nclopezo, @monttj can you please review it and eventually sign? Thanks.
@rappoccio, @imarches, @ahinzmann, @acaudron, @mmarionncern, @nhanvtran, @schoef, @ferencek, @mariadalfonso, @pvmulder, @TaiSakuma this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

matteosan1 committed:
Merged refs/pull/7946/head from repository cms-sw

cmsbuild commented:
-1 
 
Tested at: 08a91239b117eec9fe564474d3e8d7d1f8d292eb
I ran the usual tests and I found errors in the following unit tests: 
 
 ---> test runtestPhysicsToolsPatAlgos had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7946/2925/summary.html

matteosan1 committed:
fix patPhoton producer cfi

cmsbuild commented:
-1 
 
Tested at: 08a91239b117eec9fe564474d3e8d7d1f8d292eb
I ran the usual tests and I found errors in the following unit tests: 
 
 ---> test runtestPhysicsToolsPatAlgos had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7946/2936/summary.html

cmsbuild commented:
Pull request #7946 was updated. @vadler, @monttj can you please check and sign again.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7946/2936/summary.html

wmtan committed:
Root6 version numbers should be known to Root5

Dr15Jones committed:
Fix deadlock in XrdRequestManager

If the XrdCl::File is deleted while the OpenHandler is holding its
lock it can lead to a three way deadlock with locks in xrootd.

cmsbuild committed:
Merge pull request #8129 from Dr15Jones/fixDeadLockInXrdAdaptor

Fix deadlock in XrdRequestManager

davidlange6 committed:
Merge pull request #8107 from wmtan/Checksums74X

Root6 version numbers should be known to Root5

davidlange6 committed:
Merge pull request #7564 from wmtan/NewPlaybackInfo

File information in playback info

davidlange6 committed:
Merge pull request #7772 from bendavid/genstatusflags_740p7

Add GenStatusFlags

davidlange6 committed:
Merge pull request #8055 from davidlange6/pre8_perf2

Technical performance improvements for 74x

matteosan1 committed:
Merged refs/pull/7946/head from repository cms-sw

davidlange6 commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #7946 was updated. @covarell, @lveldere, @cerminar, @mommsen, @diguida, @bendavid, @civanch, @cmsbuild, @ktf, @davidlange6, @vciulli, @Dr15Jones, @cvuosalo, @mdhildreth, @deguio, @slava77, @ggovi, @vadler, @mmusich, @mulhearn, @ssekmen, @nclopezo, @danduggan, @emeschi, @thuer, @monttj, @franzoni can you please check and sign again.

cmsbuild commented:
+1
Tested at: 97fb82f499a61eefc0b42aaed9052cd4e653cc42
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7946/3250/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7946/3250/summary.html

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/7946	97fb82f

after merging manually in CMSSW_7_4_X_2015-03-11-0200 /test area test7946/, the changes are only in 
```
 PhysicsTools/PatAlgos/plugins/PATPhotonProducer.cc                 | 5 +++--
 PhysicsTools/PatAlgos/python/producersLayer1/photonProducer_cfi.py | 2 +-
 PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py             | 1 +
 3 files changed, 5 insertions(+), 3 deletions(-)

```

only miniAOD step can be affected by these changes (not reco)
jenkins shows no differences in reco quantities as expected and jenkins tests with miniAOD step completes.

diguida commented:
+1
the changes in AlCa are already merged in latest IB.

monttj commented on PhysicsTools/PatAlgos/python/producersLayer1/photonProducer_cfi.py:
```diff
@@ -3,7 +3,7 @@
     # input collection
     photonSource = cms.InputTag("gedPhotons"),
-             
+    electronSource = cms.InputTag("gedGsfElectrons"),                            
```
@matteosan1  Shouldn't we add "conversionSource" as well here?

Taejeong

matteosan1 replied:
Hi,
it doesn't matter since it is taking the default value from fillDescriptions

iDesc.add<edm::InputTag>("photonSource", edm::InputTag("no 
default"))->setComment("input collection");
iDesc.add<edm::InputTag>("electronSource", edm::InputTag("no 
default"))->setComment("input collection");
iDesc.add<edm::InputTag>("conversionSource", 
edm::InputTag("allConversions"))->setComment("input collection");


On 3/12/15 15:23, Tae Jeong Kim wrote:
>
> In PhysicsTools/PatAlgos/python/producersLayer1/photonProducer_cfi.py 
> <https://github.com/cms-sw/cmssw/pull/7946#discussion_r26304827>:
>
> > @@ -3,7 +3,7 @@
> >  patPhotons = cms.EDProducer("PATPhotonProducer",
> >      # input collection
> >      photonSource = cms.InputTag("gedPhotons"),
> > -
> > +    electronSource = cms.InputTag("gedGsfElectrons"),
>
> @matteosan1 <https://github.com/matteosan1> Shouldn't we add 
> "conversionSource" as well here?
>
> Taejeong
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/7946/files#r26304827>.
>

+++++++++++
Matteo Sani
University of California, San Diego, US.
Tel: +41227671577
CERN Office: 40-3-A02

monttj replied:
I see. thank you for the answer.

Dr15Jones commented:
+1
this really needs to be rebased since the only reason I have to sign is it added a commit from a pull request which is already in the branch.

deguio commented:
+1

monttj commented:
+1

lveldere commented:
+1

civanch commented:
+1

mulhearn commented:
+1

slava77 commented:
is this going to be merged ?
it's been a week already, and apparently it's needed for miniAOD

ggovi commented:
+1

slava77 commented:
-1 
made a rebased copy of this in #8505
that one needs less signatures (only AT, practically)

davidlange6 commented:
Thanks @slava77
